prow/cmd/deck/main.go

```diff
@@ -798,7 +798,17 @@ func handleProwJobs(ja *jobs.JobAgent, log *logrus.Entry) http.HandlerFunc {
 					jobs[i].Spec.DecorationConfig = nil
 				}
 				if set.Has(PodSpec) {
-					jobs[i].Spec.PodSpec = nil
+					// when we omit the podspec, we don't set it completely to nil
+					// instead, we set it to a new podspec that just has an empty container for each container that exists in the actual podspec
+					// this is so we can determine how many containers there are for a given prowjob without fetching all of the podspec details
+					// this is necessary for prow/cmd/deck/static/prow/prow.ts to determine whether the logIcon should link to a log endpoint or to spyglass
+					emptyContainers := []coreapi.Container{}
+					for range jobs[i].Spec.PodSpec.Containers {
```

Comment:
It isn't safe to assume the `PodSpec` field is non-nil. We should only initialize a `PodSpec` with empty containers if a PodSpec was specified. Please add a test case for this.

Reply:
Not sure if I'm just not able to find it, but it doesn't seem like we have any tests for this handler at all.  I'll go ahead and add some in main_test.go